File: js/3.js
/*
 * Solution to Project Euler Problem 3
 *
 * The prime factors of 13195 are 5, 7, 13 and 29. What is the largest prime
 * factor of the number 600851475143 ?
 */
function isPrime (x) {
  for (var i = 2; i < x; i++) {
    if (x % i === 0) {
      return false
    }
  }
  return true
}

let number = 600851475143
let divisor = 2
let primeFactors = []
while (number > 1) {
  if (isPrime(divisor)) {
    if (number % divisor === 0) {
      primeFactors.push(divisor)
      number = number / divisor
    } else {
      divisor++
    }
  } else {
    divisor++
  }
}
console.log(Math.max(...primeFactors))

File: python/3.py
"""
Solution to Project Euler problem 3.
    'What is the largest prime factor of the number 600851475143?'
"""
__author__ = 'Lucas Gracia'
__version__ = '1.0.0'
__license__ = 'MIT'

import time
import argparse


def process_arguments():
    """Processes user-entered command line arguments

    Returns:
        args.l (int) : The limit to be passed to `sum_fibs`
    """
    parser = argparse.ArgumentParser()
    parser.add_argument('-n',
                        type=int,
                        default=600851475143,
                        help='Number to be factorised')
    args = parser.parse_args()
    return args.n


def find_largest_prime_factor(number):
    """Returns largest prime factor of `number`

    Args:
        number (int) : The number to be factorised

    Returns:
        number (int) : The largest prime factor of `number`
    """
    i = 2
    while i * i < number:
        while number % i == 0:
            number = number / i
        i += 1
    return number


if __name__ == '__main__':
    START_TIME = time.time()

    NUMBER = process_arguments()
    ANSWER = find_largest_prime_factor(NUMBER)

    END_TIME = time.time()
    print("Answer:  {}\nTime: {}s".format(ANSWER, END_TIME - START_TIME))
